### PR TITLE
Use Deployment in sample-apiserver examples

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Use Deployment in sample-apiserver examples as ReplicationController is no longer the recommended way to set up replication.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/sample-apiserver/artifacts/example/deploy.yaml [hunk 1]
```diff
@@ -22,7 +23,7 @@ spec:
         # docker pull gcr.io/kubernetes-e2e-test-images/sample-apiserver:1.10
         # docker tag gcr.io/kubernetes-e2e-test-images/sample-apiserver:1.10 kube-sample-apiserver:latest
         image: kube-sample-apiserver:latest
-        imagePullPolicy: Never
+        imagePullPolicy: Always
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/deploy.yaml]: this means you cannot develop this locally and use locally built images

File: staging/src/k8s.io/sample-apiserver/artifacts/example/deploy.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-apiVersion: v1
-kind: ReplicationController
+apiVersion: apps/v1
+kind: Deployment
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/deploy.yaml]: nit: name this `deployment.yaml`

**Comment by khyveasna** [comment]
his name is in the hospital and I are going to be able to - - [just g hey I'm > his](url)

**Comment by ialidzhikov** [comment]
The tests are now ok. Could you please check again?

File: staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml [hunk 3]
```diff
@@ -22,7 +23,7 @@ spec:
         # docker pull gcr.io/kubernetes-e2e-test-images/sample-apiserver:1.17
         # docker tag gcr.io/kubernetes-e2e-test-images/sample-apiserver:1.17 kube-sample-apiserver:latest
         image: kube-sample-apiserver:latest
-        imagePullPolicy: Never
+        imagePullPolicy: IfNotPresent
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml]: given `kube-sample-apiserver:latest` is not expected to be able to be pulled from public registries (it depends on being present and tagged according to the instructions above), shouldn't we leave this as "Never"?

Comment (ialidzhikov) [reply staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml]: Yep, changed to `Never`.

**Pull request merged**